### PR TITLE
689 convert to ophyd async ipin

### LINKED ISSUE #689: Convert to ophyd-async: ipin

### PR DESCRIPTION
**Pull request by stan-dot**
Fixes #689

### Instructions to reviewer on how to test:
1. run dodal connect i04 
2. make sure the device is connected
3. make sure the device is readable in a run (plans in [mx_bluesky](https://github.com/DiamondLightSource/mx_bluesky))

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

File: src/dodal/devices/ipin.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
 
 
-class IPin(Device):
+class IPinAsync(StandardReadable):
     """Simple device to get the ipin reading"""
 
-    reading = Cpt(EpicsSignalRO, "I", kind=Kind.hinted)
+    def __init__(self, prefix: str, name: str = "") -> None:
+        pin_readback = epics_signal_r(float, prefix + "I")
+        self.add_readables([pin_readback], wrapper=HintedSignal)
```
Comment (DominicOram) [inline src/dodal/devices/ipin.py:10]: Nit: I think the context manager syntax is slightly nicer:
```suggestion
        with self.add_children_as_readables(wrapper=HintedSignal):
            pin_readback = epics_signal_r(float, prefix + "I")
```
But not too bothered in the case of only one child like this

File: src/dodal/devices/ipin.py [hunk 2]
```diff
@@ -1,8 +1,11 @@
-from ophyd import Component as Cpt
-from ophyd import Device, EpicsSignalRO, Kind
+from ophyd_async.core import HintedSignal, StandardReadable
+from ophyd_async.epics.signal import epics_signal_r
 
 
-class IPin(Device):
+class IPinAsync(StandardReadable):
```
Comment (DominicOram) [inline src/dodal/devices/ipin.py:5]: Should: I think given that every device will be an `Async` one in the next month or so the word `Async` here is redundant, we can just remove it.

**Review by DominicOram — CHANGES_REQUESTED** [review]
Great, thank you. One comment on the naming, otherwise good.

File: src/dodal/devices/ipin.py (hunk 2 above)
Comment (stan-dot) [reply src/dodal/devices/ipin.py:5]: ok

File: src/dodal/devices/ipin.py (hunk 1 above)
Comment (stan-dot) [reply src/dodal/devices/ipin.py]: fixed

File: src/dodal/devices/ipin.py (hunk 2 above)
Comment (stan-dot) [reply src/dodal/devices/ipin.py]: fixed

**Review by DiamondJoseph — APPROVED** [review]
LGTM

**Comment by DiamondJoseph** [comment]
@stan-dot please squash and merge and rewrite the squashed commits to something sensible

**Pull request merged**